official/resnet/resnet_run_loop.py

```diff
@@ -266,8 +277,19 @@ def exclude_batch_norm(name):
     if multi_gpu:
       optimizer = tf.contrib.estimator.TowerOptimizer(optimizer)
 
+    if loss_scale != 1:
+      # When computing fp16 gradients, often intermediate tensor values are
+      # so small, they underflow to 0. To avoid this, we multiply the loss by
+      # loss_scale to make these tensor values loss_scales times bigger.
+      scaled_grad_vars = optimizer.compute_gradients(loss * loss_scale)
+      unscaled_grad_vars = [(grad / loss_scale, var)
+                            for grad, var in scaled_grad_vars]
+      minimize_op = optimizer.apply_gradients(unscaled_grad_vars, global_step)
```

Comment:
Extract to separate function and test? This seems like a good candidate for something the optimizer should handle itself. Consider adding to core? I would suggest you add a bug, self-assign, and see if this is something that is generalizable and of general interest.

Reply:
Since we are not formalizing the mixed precision code at this time, my inclination is to leave it as is and consider doing this if it gets moved to a util. Seem reasonable?